src/string/strnlen.c

```diff
@@ -1,7 +1,7 @@
 #include <string.h>
 
-size_t strnlen(const char *s, size_t n)
-{
-	const char *p = memchr(s, 0, n);
+size_t strnlen(const char *s : itype(_Nt_array_ptr<const char>) count(n), size_t n)
+_Checked{
+	_Nt_array_ptr<const char> p : count(n) = _Dynamic_bounds_cast<_Nt_array_ptr<const char>>(memchr(s, 0, n), count(n));
```

Comment:
Dynamic_bounds_cast may be unsafe. I think here, we should use Array_ptr instead of nt_array_ptr.